iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java

```diff
@@ -69,14 +78,37 @@ public StringLiteral stringLiteral(SyntaxToken token) {
     return new StringLiteralImpl(token);
   }
 
+  // Ignore unused method parameters
+  @SuppressWarnings("java:S1172")
+  public ResourceDeclaration resourceDeclaration(
+    SyntaxToken keyword,
+    Identifier identifier,
+    InterpolatedString type,
+    Optional<SyntaxToken> existing,
+    SyntaxToken equalsSign,
+    ObjectExpression objectExpression,
+    SyntaxToken endOfLine) {
+    return new ResourceDeclarationImpl(keyword, identifier, type, existing.orNull(), equalsSign, objectExpression);
```

Comment:
Let's store the EOL token in the tree. I think about a super simple rule like trailing white spaces.